account_bank_statement_import_camt_oca/models/parser.py

```diff
@@ -60,6 +61,20 @@ def parse_transaction_details(self, ns, node, transaction):
                 './ns:AddtlNtryInf',
                 './ns:Refs/ns:InstrId',
             ], transaction, 'name', join_str='\n')
+        if self._get_journal_use_ref_as_label_stmt_import():
+            self.add_value_from_node(
+                ns, node, [
+                    './ns:RmtInf/ns:Strd/ns:CdtrRefInf/ns:Ref',
+                    './ns:Refs/ns:EndToEndId',
+                    './ns:Ntry/ns:AcctSvcrRef',
+                ], transaction, 'name2')
+            if (
+                transaction['name'] != '/'
+                and transaction['name2']
+                and transaction['name'] != transaction['name2']
+            ):
+                transaction['name'] += ' ' + transaction['name2']
+                del transaction['name2']
```

Comment:
Maybe you can add a comment with an example to explain why you need to create and delete the `name2` node.